spec/multi_file_precompile_hook_spec.rb

```diff
@@ -0,0 +1,24 @@
+require_relative './spec_helper.rb'
+
+describe Mumukit::Templates::MultiFilePrecompileHook do
+  class SomeMultiFilePrecompileHook < Mumukit::Templates::MultiFilePrecompileHook
+    def main_file
+      'main.js'
+    end
+
+    def consolidate(main_content, files)
+      main_content.gsub(/(import\(\"([\w\.]+)\"\);)/) { files[$2] }
+    end
+  end
+
+  let(:request) {
+    req "main.js": 'console.log("hello");   import("anotherFile.js");   console.log("!!");',
```

Comment:
@rodri042 why not passing each file within a dictionary? e.g.:

```ruby
req content: { "main.js": "...", "another.js": "..." }, test: "..."
```

or

```ruby
req files: { "main.js": "...", "another.js": "..." }, test: "..."
```

That way, in order to detect the files of the multifile request, you don't have to trust the extension. Otherwise you can get into problems when dealing with certain file formats that don't have extensions - think of Makefiles or Dockerfiles. 

I know we are using extensions to determine MIME type when using context highlight, but if possible I'd want to avoid relaying on that in more deep decisions about the architecture, since they can't changed so easily .

Reply:
PS: for example, in the UI we are assuming the dot-convention 

https://github.com/mumuki/mumuki-laboratory/pull/1150/files#diff-7964aafebf8eaa6037d10503533a4b4dR195

https://github.com/mumuki/mumuki-laboratory/pull/1150/files#diff-7964aafebf8eaa6037d10503533a4b4dR120

However, although is may be problematic in the future, it is still easier to change latter than the communication protocol between the runners.